File: 백준/Silver/1874. 스택 수열/스택 수열.py
N = int(input())

count = 0
stack = []
result = []
cant_make = False

for _ in range(N):
    x = int(input())

    while count < x:
        count += 1
        stack.append(count)
        result.append('+')

    if stack[-1] == x:
        stack.pop()
        result.append('-')
    else:
        cant_make = True
        break

if cant_make:
    print("NO")
else:
    print("\n".join(result))

File: 백준/Silver/1874. 스택 수열/스택 수열.js
const input = require('fs').readFileSync(process.version === 'v18.17.1' ? 'input.txt' : '/dev/stdin')
  .toString().trim().split("\n");

const solution = () => {
  const N = +input[0];
  const numbers = input.slice(1).map(Number);

  let stack = [];
  let current = 1;
  let result = [];

  for (let num of numbers) {
    while (current <= num) {
      stack.push(current);
      result.push('+');
      current++;
    }

    if (stack[stack.length - 1] === num) {
      stack.pop();
      result.push('-');
    } else {
      console.log('NO');
      return;
    }
  }

  console.log(result.join('\n'));
};

solution();
